base/float.jl

```diff
@@ -386,10 +386,35 @@ end
 end
 
 # fused multiply-add
-fma_libm(x::Float32, y::Float32, z::Float32) =
-    ccall(("fmaf", libm_name), Float32, (Float32,Float32,Float32), x, y, z)
-fma_libm(x::Float64, y::Float64, z::Float64) =
-    ccall(("fma", libm_name), Float64, (Float64,Float64,Float64), x, y, z)
+function fma_libm(x::Float32, y::Float32, z::Float32)
+    r = ccall(("fmaf", libm_name), Float32, (Float32,Float32,Float32), x, y, z)
+    # This is necessary on 32-bit Intel Linux, probably on OS X, and
+    # maybe on Windows
+    # TODO: Disable this for non-Intel systems
+    if is(Int, Int32)
```

Comment:
I don't have a direct access to a 32-bit machine, only via Fedora builders. AFAICS in Make.inc, `ARCH` is extracted from `$(CC) -dumpmachine`, so my statement that we may get any option accepted by `-mtune` looks exaggerated as long as the user doesn't override this variable. Still, there are quite a few different cases handled in the Makefiles: https://github.com/JuliaLang/julia/blob/master/Make.inc#L476-496 But since all unknown values currently lead to a failure, it should be fine not to handle them.

If I were you, I'd create a new variable at these lines, and write it to `base/build_h.jl`. That way, it will evolve naturally with the Makefiles if something changes in the future.